### DISCUSSION TITLE
getShopReceiptTransactionsByReceipt - returns information about any receipt_id

### DISCUSSION THREAD

**Discussion by robpal** [etsy/open-api#979] [Show and tell]
I think there should be a major bug.

`/v3/application/shops/{shop_id}/receipts/{receipt_id}/transactions`

Because now with this endpoint I can get information about any `receipt_id` on ETSY platform.

It doesn't check if `receipt_id` is associated with `shop_id`.

**Comment by Gareth064** [comment]
I no it's a bit dodgy, but can you give me a receipt ID from another shop and I can test it using my test app to see if it returns anything. I will use my shop_id wit your receipt_id

**Reply by colesnic89** [reply]
@Gareth064  confirm. If I query receipt from other shop I have as response 
`{
    "error": "Shop_Receipt2 with PK shop_id = 12569287 AND receipt_id = 47291010 does not exist"
}`

but if I query transaction for this receipt I have 
```{
    "count": 1,
    "results": [
        {
            "transaction_id": 71185853,
            "title": "Your Choice of Five Patterns - Buy More and Save - PDF Sale",
            "description": "Your choice of any five patterns listed in my shop.\r\n\r\nYou have permission to sell finished products - please include a link to my Etsy shop in your product listing:\r\n\r\nwww.etsy.com/shop/melodysmakings\r\n\r\n\r\n\r\nSpecify your choices at purchase.\r\n\r\n\r\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n\r\nLike my Facebook page - Melody&#39;s Makings for special offers, giveaways, pattern testing opportunities, and more!",
            "seller_user_id": 12431079,`

**Reply by colesnic89** [reply]
@Gareth064 my shop receiptId as example `2868329653`

**Reply by Gareth064** [reply]
ok checking now.

**Reply by Gareth064** [reply]
Ok first test was to make sure I couldn't get to the Order on the Web :P
![image](https://user-images.githubusercontent.com/22307223/234855979-553e3cdd-fd19-4e88-86e4-9510f4ef97c7.png)

I get the same error when trying to get the receipt, using my Shop_ID
```json
{
  "ResponseCode": 404,
  "Success": false,
  "Data": null,
  "Message": "Shop_Receipt2 with PK shop_id = {shop_id} AND receipt_id = 2868329653 does not exist: ",
  "ResponseHeaders": {
    "LimitPerSecond": 10,
    "RemainingThisSecond": 9,
    "LimitPerDay": 10000,
    "RemainingToday": 9986,
    "EtsyRequestUuid": "EufuRmoPjYNTqreJcZZcNNb6VAc4"
  },
  "RequestedResource": null
}
```

**Reply by Gareth064** [reply]
Here are my overall Tests when using receipt_id `2868329653` and transaction_id `71185853`

[getShopReceipt](https://developers.etsy.com/documentation/reference/#operation/getShopReceipt)
![image](https://user-images.githubusercontent.com/22307223/234857798-68678ee5-f556-456f-9bd6-ffd20740c820.png)

[getShopReceiptTransactionsByReceipt](https://developers.etsy.com/documentation/reference/#operation/getShopReceiptTransactionsByReceipt)
![image](https://user-images.githubusercontent.com/22307223/234858146-6161d266-38fb-4555-aa73-c7ba89b0284e.png)

[getShopReceiptTransaction](https://developers.etsy.com/documentation/reference/#operation/getShopReceiptTransaction)
![image](https://user-images.githubusercontent.com/22307223/234858443-83685db8-fa4a-4432-a525-eb04b6df17d4.png)

Then a bonus test using the transaction_id returned in my second test above
![image](https://user-images.githubusercontent.com/22307223/234858816-45850c05-ee2d-4ec7-9066-49055cf16a5d.png)

So the final question to ask is WHY does [getShopReceiptTransactionsByReceipt](https://developers.etsy.com/documentation/reference/#operation/getShopReceiptTransactionsByReceipt) return a result for a receipt_id which is not related to the shop_id?

**Reply by Gareth064** [reply]
I will email Etsy Dev support about this. But for now @nateswart just giving you a heads up about it. This seems pretty urgent as the transaction returned has all the id's of buyers, shops, etc.... and although no shipping info is present, the personalisation text could contain sensitive information

**Reply by robpal** [reply]
At least you can't get more details about buyer, shipping and etc only ID :) I could be a major data leak

**Reply by Gareth064** [reply]
100%

**Comment by nateswart** [comment]
Thank you for bringing this to our attention.
